play-services-auth/src/main/java/com/google/android/gms/auth/api/identity/BeginSignInRequest.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-FileCopyrightText: 2023 microG Project Team
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.google.android.gms.auth.api.identity;
+
+import org.microg.safeparcel.AutoSafeParcelable;
+
+public class BeginSignInRequest extends AutoSafeParcelable {
```

Comment:
Please make use of the new safe-parcel-processor wherever easily possible.
1. Ensure that `annotationProcessor project(':safe-parcel-processor')` is added to `dependencies` of corresponding `build.gradle`.
2. Make the class extend `AbstractSafeParcelable` instead of `AutoSafeParcelable`
3. Add `@SafeParcelable.Class` annotiation to the class
4. Make sure the `CREATOR` is defined as `public static final SafeParcelableCreatorAndWriter<XXX> CREATOR = findCreator(XXX.class);`
5. Add `writeToParcel` to the class:
  ```java
  @Override
  public void writeToParcel(@NonNull Parcel dest, int flags) {
      CREATOR.writeToParcel(this, dest, flags);
  }
  ```
6. If necessary, make private fields package-visible or provide a getter (referenced via `getterName`) and constructor parameter for them, tagged with `@Param`.
7. If necessary, tag the constructor to be used when deserializing with `@Constructor` (if no constructor is declared, the default constructor will be used).

Using this system severely reduces the use of reflection at runtime, which can significantly improve performance as it allows for better optimizations.

Reply:
I don’t understand it very well here. AutoSafeParcelable inherits from AbstractSafeParcelable and uses SafeParcelableCreatorAndWriter.

Reply:
`AutoSafeParcelable` is using reflection. Reflection isn't good for performance as it can't be optimized. When using the system based on the annotation processor, as explained above, reflection is not used during (de)serialization. I outlined how to use annotation processor above, you can also take a look at other classes that directly inherit from `AbstractSafeParcelable` without using `AutoSafeParcelable`.

Reply:
Oh oh, I changed it